subprojects/android-test/ui-testing-core/src/main/kotlin/com/avito/android/test/espresso/action/TouchOutsideAction.kt

```diff
@@ -0,0 +1,88 @@
+package com.avito.android.lib.actions
+
+import android.graphics.Rect
+import android.os.SystemClock
+import android.view.InputDevice
+import android.view.MotionEvent
+import android.view.MotionEvent.PointerProperties
+import android.view.View
+import android.view.ViewConfiguration
+import android.view.ViewGroup
+import androidx.test.espresso.UiController
+import androidx.test.espresso.ViewAction
+import androidx.test.espresso.action.CoordinatesProvider
+import androidx.test.espresso.matcher.ViewMatchers
+import androidx.test.platform.app.InstrumentationRegistry
+import androidx.test.uiautomator.Configurator
+import androidx.test.uiautomator.UiDevice
+import org.hamcrest.Matcher
+
+class TouchOutsideAction : ViewAction {
+
+    override fun getConstraints(): Matcher<View> = ViewMatchers.isAssignableFrom(ViewGroup::class.java)
```

Comment:
Is this a harder constraint for dialog?

Reply:
Change to `View::class.java` because, as you mentioned, action uses for View, not for Dialog itself